engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -600,6 +600,19 @@ protected AgentAttache notifyMonitorsOfConnection(final AgentAttache attache, fi
             // return the attache instead of null, even it is disconnectede
             handleDisconnectWithoutInvestigation(attache, Event.AgentDisconnected, true, true);
         }
+        if (answer instanceof ReadyAnswer) {
+            ReadyAnswer readyAnswer = (ReadyAnswer)answer;
+            Map<String, String> detailsMap = readyAnswer.getDetailsMap();
+            if (detailsMap != null) {
+                String uefiEnabled = detailsMap.get(Host.HOST_UEFI_ENABLE);
+                s_logger.debug(String.format("Got HOST_UEFI_ENABLE [%s] for hostId [%s]:", uefiEnabled, host.getUuid()));
+                if (uefiEnabled != null) {
+                    _hostDao.loadDetails(host);
+                    host.getDetails().put(Host.HOST_UEFI_ENABLE, uefiEnabled);
+                    _hostDao.saveDetails(host);
```

Comment:
What about checking first and persisting only in case it was not present or the value changed?